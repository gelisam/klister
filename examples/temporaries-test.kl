#lang "n-ary-app.kl"

(import (shift "n-ary-app.kl" 1))
(import (shift "lispy-do.kl" 1))
(import (shift "list-datatype.kl" 1))
(import (shift "pair-datatype.kl" 1))
(import (shift "quasiquote.kl" 1))
(import (shift "temporaries.kl" 1))

(define-macros
  ([tester
    (lambda (stx)
      (do (<- tmps (make-temporaries (:: (unit) (:: (unit) (:: (unit) (nil))))))
          (case tmps
            [(:: a as)
             (case as
               [(:: b bs)
                (case bs
                  [(:: c cs)
                   (pure (quasiquote/loc stx (lambda (,(fst a) ,(fst b) ,(fst c)) ,(fst b))))])])])))]))

(example ((tester) 'a 'b 'c))
