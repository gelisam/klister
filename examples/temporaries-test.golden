#[temporaries-test.kl:22.24-22.25]<b> : Syntax
